### PR TITLE
Update vancouver-imperial-college-london.csl

### PR DESCRIPTION
**Pull request by simomac**
Update to the Vancouver citation style from Imperial College London Library reflecting our latest guidelines.

### CODE REVIEW

File: vancouver-imperial-college-london.csl [hunk 1]
```diff
@@ -256,8 +254,16 @@
   <macro name="journal-location">
     <choose>
       <if type="article-journal article-magazine review review-book" match="any">
-        <text variable="volume"/>
-        <text variable="issue" prefix="(" suffix=")"/>
+        <choose>
```
Comment (adam3smith) [inline vancouver-imperial-college-london.csl:257]: this seems unnecessarily complex -- what is it designed to fix/address?

File: vancouver-imperial-college-london.csl [hunk 2]
```diff
@@ -40,24 +44,29 @@
...
     <choose>
-      <if type="report" match="none">
+      <if type="report article article-journal" match="none">
         <group delimiter=": " suffix=";">
           <choose>
             <if type="thesis">
               <text variable="publisher-place" prefix="[" suffix="]"/>
             </if>
-            <else-if type="speech"/>
+            <else-if type="speech article-journal report" match="any"/>
```
Comment (adam3smith) [inline vancouver-imperial-college-london.csl:69]: so just to confirm: reports would get neither publisher nor publisher-place? That seems unusual?

File: vancouver-imperial-college-london.csl [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (adam3smith) [inline vancouver-imperial-college-london.csl:1]: revert please

File: vancouver-imperial-college-london.csl [hunk 4]
```diff
@@ -77,10 +85,9 @@
     <choose>
       <if variable="URL DOI" match="any">
         <group delimiter=" ">
-          <text term="available at" text-case="capitalize-first" suffix=":"/>
           <choose>
             <if variable="DOI">
-              <text variable="DOI" prefix="doi:"/>
+              <text variable="DOI" prefix="doi:" suffix="."/>
```
Comment (adam3smith) [inline vancouver-imperial-college-london.csl:90]: shouldn't the prefix be "https://doi.org" in line with the guidelines?

**Review by adam3smith — CHANGES_REQUESTED** [review]
Some small change requests and some questions. Thanks!

**Comment by adam3smith** [comment]
@simomac will you be able to make the requested changes? Thanks in advance!

**Comment by simomac** [comment]
Hi Sebastian, working on the changes, amends will be with you by end of today.

Best
Simon

File: vancouver-imperial-college-london.csl (hunk 2 above)
Comment (simomac) [reply vancouver-imperial-college-london.csl:69]: Yes, that's been confirmed by our Liason librarians.: '...I think they have a point here, the publisher and place should be included for reports. However the print guide and web guide doesn’t include this information so for consistency I think we should stick with the status quo. When we look at referencing again next year we can decide whether to change this across the board....'

File: vancouver-imperial-college-london.csl (hunk 4 above)
Comment (simomac) [reply vancouver-imperial-college-london.csl:90]: No, this should stay just as ‘doi’, we don’t include the https: part

File: vancouver-imperial-college-london.csl (hunk 1 above)
Comment (simomac) [reply vancouver-imperial-college-london.csl:257]: This is designed to distinguish the issue number , where it exists, from the volume by placing it in parenthesis

File: vancouver-imperial-college-london.csl (hunk 3 above)
Comment (simomac) [reply vancouver-imperial-college-london.csl]: done

File: vancouver-imperial-college-london.csl (hunk 1 above)
Comment (adam3smith) [reply vancouver-imperial-college-london.csl:257]: But that's not what it's doing: the issue is already in parentheses with the previous code. For some reason you're checking if volume and issue are numbers? That seems particularly problematic for issue, which you also see as "Fall", e.g.

File: vancouver-imperial-college-london.csl (hunk 4 above)
Comment (adam3smith) [reply vancouver-imperial-college-london.csl:90]: FWIW, your guidance is contradictory here. You have some examples with doi: but also 


And doi: is discouraged by CrossRef and the DOI foundation. Any chance you could change/unify that in the guidance?

**Comment by adam3smith** [comment]
Thanks! Still have questions about two of these

File: vancouver-imperial-college-london.csl (hunk 1 above)
Comment (simomac) [reply vancouver-imperial-college-london.csl]: Hi, this has been simplified.

File: vancouver-imperial-college-london.csl (hunk 4 above)
Comment (simomac) [reply vancouver-imperial-college-london.csl]: changed o https/..... as requested

**Comment by adam3smith** [comment]
Thanks!

**Pull request merged**